### PR TITLE
feat: renewal auto operations tab

### PR DESCRIPTION
**Pull request by bimalgrg519**

### CODE REVIEW

**Comment by cre8ivejp** [comment]
@bimalgrg519, the calendar doesn't show if you do the following steps.

1- Click on New Operation
2- Click on the Kill Switch tab
3- Click on Event Rate
4- Click on Enable tab
5- Click on the Schedule

Can you take a look?

**Comment by bimalgrg519** [comment]
@cre8ivejp  resolved.

**Comment by bimalgrg519** [comment]
Screenshot 2023-09-07 at 16 15 03
@cre8ivejp  I have fixed following issues as per our last discussion.

1. Display the current event rate as 0% until the minimum counts reached.
2. Add `Total Count Events` in the UI.
3. Add a tootltip in the current event rate info icon.
4. Make the graph according to the threshold percentage.

**Comment by cre8ivejp** [comment]
@bimalgrg519, can you round the percentage?
8%, 16%, 24%, ...

Also, please update the field name percentage on the create page to `Threshold`.

**Comment by cre8ivejp** [comment]
@bimalgrg519, can you fix this?
When I click to edit an event rate, the radio button focuses on the wrong button. The focus should be on the event rate.

Screenshot 2023-09-08 at 11 07 25 AM

**Comment by bimalgrg519** [comment]
@cre8ivejp  Above issues has been resolved.

**Comment by cre8ivejp** [comment]
@bimalgrg519, please fix the translations on the form.

Conditions -> 条件
Threshold -> スレッショルド

Also, the data format on the left side should be the same as on the right for scheduled operations. Use the created info.

Screenshot 2023-09-11 at 10 17 02 AM

**Comment by cre8ivejp** [comment]
@bimalgrg519, can you change to list 11 points when you create the percentage graph?
I want you to divide it by 10 but not include the zero in the calculation. So it should have 11 points until the threshold.

E.g.
0-5-10-15-20-25-30-35-40-45-50

Screenshot 2023-09-11 at 10 26 02 AM

**Comment by cre8ivejp** [comment]
Also, please these translations.

Enable -> 有効
Kill Switch -> キルスイッチ
Schedule -> スケジュール
Event Rate -> イベントレート
Start date -> 開始日時

**Comment by cre8ivejp** [comment]
@bimalgrg519, please keep the original primary button color.
The one you used is for the new design.

Screenshot 2023-09-11 at 11 28 37 AM

**Comment by bimalgrg519** [comment]
@cre8ivejp Above four issues has been resolved. Please have a look.

**Comment by cre8ivejp** [comment]
@bimalgrg519, did you push the changes?

**Comment by bimalgrg519** [comment]
> @bimalgrg519, did you push the changes?

Sorry. Please take a look now.

**Comment by cre8ivejp** [comment]
@bimalgrg519, when the event rate is zero in the completed tab, it shows as "0%", but on the active tab, it only shows "0".
Can you add the percentage?

Screenshot 2023-09-11 at 8 05 32 PM

**Comment by bimalgrg519** [comment]
> @bimalgrg519, when the event rate is zero in the completed tab, it shows as "0%", but on the active tab, it only shows "0". Can you add the percentage?
> 
> Screenshot 2023-09-11 at 8 05 32 PM

@cre8ivejp  Percentage added.

**Comment by bimalgrg519** [comment]
@cre8ivejp  Translations added.

File: ui/web-v2/apps/admin/src/components/FeatureAutoOpsRulesForm/index.tsx [hunk 1]
```diff
@@ -38,39 +71,536 @@ export const ClauseType: ClauseTypeMap = {
...
+            const { typeUrl } = rule.clausesList[0].clause;
+            const type = typeUrl.substring(typeUrl.lastIndexOf('/') + 1);
+            return type === ClauseType.EVENT_RATE && !rule.triggeredAt;
+          })
+          .map((rule) => rule.id);
+
+        if (ids.length > 0) {
+          dispatch(
+            listOpsCounts({
+              environmentNamespace: currentEnvironment.namespace,
```
Comment (masaaania) [inline ui/web-v2/apps/admin/src/components/FeatureAutoOpsRulesForm/index.tsx:129]:
```suggestion
              environmentNamespace: currentEnvironment.id,
```

File: ui/web-v2/apps/admin/src/components/FeatureAutoOpsRulesForm/index.tsx [hunk 2]
```diff
@@ -38,39 +71,536 @@ export const ClauseType: ClauseTypeMap = {
...
+            })
+          );
+        }
+      }
+    }, [autoOpsRules]);
+
+    const handleClose = useCallback(() => {
+      reset();
+      history.replace({
+        pathname: `${PAGE_PATH_ROOT}${currentEnvironment.id}${PAGE_PATH_FEATURES}/${featureId}${PAGE_PATH_FEATURE_AUTOOPS}`,
```
Comment (masaaania) [inline ui/web-v2/apps/admin/src/components/FeatureAutoOpsRulesForm/index.tsx:140]:
```suggestion
        pathname: `${PAGE_PATH_ROOT}${currentEnvironment.urlCode}${PAGE_PATH_FEATURES}/${featureId}${PAGE_PATH_FEATURE_AUTOOPS}`,
```

File: ui/web-v2/apps/admin/src/components/OperationAddUpdateForm/index.tsx [hunk 3]
```diff
@@ -0,0 +1,900 @@
...
+            const command = new ChangeOpsEventRateClauseCommand();
+            command.setId(autoOpsRule.clausesList[0].id);
+            command.setOpsEventRateClause(
+              createOpsEventRateClause(data.eventRate)
+            );
+            changeOpsEventRateClauseCommands.push(command);
+          }
+
+          const param: UpdateAutoOpsRuleParams = {
+            environmentNamespace: currentEnvironment.namespace,
```
Comment (masaaania) [inline ui/web-v2/apps/admin/src/components/OperationAddUpdateForm/index.tsx:223]:
```suggestion
            environmentNamespace: currentEnvironment.id,
```

File: ui/web-v2/apps/admin/src/components/OperationAddUpdateForm/index.tsx [hunk 4]
```diff
@@ -0,0 +1,900 @@
...
+    mode: 'onChange',
+  });
+
+  const onSubmit = (data) => console.log(data);
+
+  const handleCreateGoal = useCallback(
+    async (data) => {
+      dispatch(
+        createGoal({
+          environmentNamespace: currentEnvironment.namespace,
```
Comment (masaaania) [inline ui/web-v2/apps/admin/src/components/OperationAddUpdateForm/index.tsx:748]:
```suggestion
          environmentNamespace: currentEnvironment.id,
```

File: ui/web-v2/apps/admin/src/pages/feature/detail.tsx [hunk 5]
```diff
@@ -71,15 +82,25 @@ export const FeatureDetailPage: FC = memo(() => {
           id: featureId,
         })
       );
+      dispatch(
+        listAutoOpsRules({
+          featureId: featureId,
+          environmentNamespace: currentEnvironment.namespace,
```
Comment (masaaania) [inline ui/web-v2/apps/admin/src/pages/feature/detail.tsx:88]:
```suggestion
          environmentNamespace: currentEnvironment.id,
```

**Review by masaaania — CHANGES_REQUESTED** [review]
Sorry, I merged the change from [environment V1](https://github.com/bucketeer-io/bucketeer/blob/36b5c5fc397ea42b5cf81df6dcb7d76f82520f56/ui/web-v2/apps/admin/src/proto/environment/environment_pb.d.ts#L41-L52) to [environment V2](https://github.com/bucketeer-io/bucketeer/blob/36b5c5fc397ea42b5cf81df6dcb7d76f82520f56/ui/web-v2/apps/admin/src/proto/environment/environment_pb.d.ts#L89-L100) in this PR.
https://github.com/bucketeer-io/bucketeer/pull/528
Please add the changes I suggested 🙏 .

File: ui/web-v2/apps/admin/src/components/FeatureAutoOpsRulesForm/index.tsx (hunk 2 above)
Comment (bimalgrg519) [reply ui/web-v2/apps/admin/src/components/FeatureAutoOpsRulesForm/index.tsx:140]: @masaaania  There is no `urlCode` in `currentEnvironment`.

Comment (bimalgrg519) [reply ui/web-v2/apps/admin/src/components/FeatureAutoOpsRulesForm/index.tsx:140]: Here is the type for currentEnvironment
Screenshot 2023-09-28 at 12 58 05

Comment (cre8ivejp) [reply ui/web-v2/apps/admin/src/components/FeatureAutoOpsRulesForm/index.tsx:140]: @bimalgrg519, please rebase your project from the main branch and regenerate the proto files again.

Comment (bimalgrg519) [reply ui/web-v2/apps/admin/src/components/FeatureAutoOpsRulesForm/index.tsx:140]: @cre8ivejp  Thank you

**Comment by bimalgrg519** [comment]
@masaaania  Changes have been added.

**Review by masaaania — APPROVED** [review]
Thank you!!

**Review by cre8ivejp — APPROVED** [review]
Thank you!

**Pull request merged**